sentry_sdk/integrations/sanic.py

```diff
@@ -297,51 +293,47 @@ async def _legacy_handle_request(self, request, *args, **kwargs):
 def _legacy_router_get(self, *args):
     # type: (Any, Union[Any, Request]) -> Any
     rv = old_router_get(self, *args)
-    hub = Hub.current
-    if hub.get_integration(SanicIntegration) is not None:
+    if sentry_sdk.get_client().get_integration(SanicIntegration) is not None:
         with capture_internal_exceptions():
-            with hub.configure_scope() as scope:
-                if SanicIntegration.version and SanicIntegration.version >= (21, 3):
-                    # Sanic versions above and including 21.3 append the app name to the
-                    # route name, and so we need to remove it from Route name so the
-                    # transaction name is consistent across all versions
-                    sanic_app_name = self.ctx.app.name
-                    sanic_route = rv[0].name
-
-                    if sanic_route.startswith("%s." % sanic_app_name):
-                        # We add a 1 to the len of the sanic_app_name because there is a dot
-                        # that joins app name and the route name
-                        # Format: app_name.route_name
-                        sanic_route = sanic_route[len(sanic_app_name) + 1 :]
-
-                    scope.set_transaction_name(
-                        sanic_route, source=TRANSACTION_SOURCE_COMPONENT
-                    )
-                else:
-                    scope.set_transaction_name(
-                        rv[0].__name__, source=TRANSACTION_SOURCE_COMPONENT
-                    )
+            scope = Scope.get_isolation_scope()
+            if SanicIntegration.version and SanicIntegration.version >= (21, 3):
+                # Sanic versions above and including 21.3 append the app name to the
+                # route name, and so we need to remove it from Route name so the
+                # transaction name is consistent across all versions
+                sanic_app_name = self.ctx.app.name
+                sanic_route = rv[0].name
+
+                if sanic_route.startswith("%s." % sanic_app_name):
+                    # We add a 1 to the len of the sanic_app_name because there is a dot
+                    # that joins app name and the route name
+                    # Format: app_name.route_name
+                    sanic_route = sanic_route[len(sanic_app_name) + 1 :]
+
+                scope.set_transaction_name(
+                    sanic_route, source=TRANSACTION_SOURCE_COMPONENT
+                )
+            else:
+                scope.set_transaction_name(
+                    rv[0].__name__, source=TRANSACTION_SOURCE_COMPONENT
+                )
 
     return rv
 
 
 def _capture_exception(exception):
     # type: (Union[Tuple[Optional[type], Optional[BaseException], Any], BaseException]) -> None
-    hub = Hub.current
-    integration = hub.get_integration(SanicIntegration)
+    client = sentry_sdk.get_client()
+    integration = client.get_integration(SanicIntegration)
     if integration is None:
```

Comment:
We will be able to use #2893 here

Reply:
We can do one big sweep to change this in all integrations where it makes sense once the PR is merged. There are quite a few places also in the integrations that have already been merged.